Ricochet Projectile

Isaelus opened the pull request:
This is a projectile that moves fast and bounces up to 5 times before it despawns. When it hits an entity, it despawns as well.

Isaelus committed:
Bounces 5 times, moves fast(?)

Kainkun committed:
Merge branch 'main' into RicochetProjectile

Kainkun reviewed (changes requested):
it's not compiling. ask Jacob if you need help setting it up

BUNN1E5 commented:
So a way you can look at the ProjectileWeapon Script, is that it is the Gun and that's it, if you want it to shoot anything you have to make the bullets yourself

The Weapon itself shouldn't be getting shot

Isaelus committed:
Merge branch 'main' into RicochetProjectile

Isaelus committed:
Fixed most errors of Ricochet and RicochetBullet.

Ricochet Bullet - couldn't use GetComponent b/c it had no reference and fixed it with other.gameObject.GetComponent
Ricochet - transform wasn't in context so I just made a Transform reference.

Kainkun committed:
Merge branch 'main' into RicochetProjectile

Isaelus committed:
Merge branch 'main' into RicochetProjectile

Isaelus committed:
Merge branch 'RicochetProjectile' of https://github.com/AztecGameLab/LongTermGame5 into RicochetProjectile

Isaelus committed:
Merge branch 'main' into RicochetProjectile

Commit:
Merge branch 'main' of https://github.com/AztecGameLab/LongTermGame5 into RicochetProjectile

BUNN1E5 commented on Assets/Scripts/Ricochet.cs:
```diff
@@ -0,0 +1,30 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+[CreateAssetMenu(fileName = "RicochetWeapon", menuName = "RicochetWeapon")]
```
Please change this to 
`[CreateAssetMenu(fileName = "Ricochet", menuName = "LTG5/Weapons/Ricochet")]`

BUNN1E5 commented on Assets/Scripts/RicochetBullet.cs:
```diff
@@ -0,0 +1,38 @@
+
+    private void OnCollisionEnter2D(Collision2D other)
+    {
+        if (other.gameObject.layer == layer.Entity)
```
I'm not exactly sure if we really are using layers. So instead of checking layers what you can do is `Entity entity = other.gameObject.GetComponent<Entity>();` and then check if entity is null, from there we know that it isn't an entity, and the proceed to do the reflect code

BUNN1E5 reviewed (changes requested):
I'm not exactly certain how to test your code, I'm not seeing the scene that you used, can you provide instructions on how to test it.

I can see that you were confused about the Fire/Charge/OnAimChange functions, normally I check the Pull request often, but this was a busy week please do not be afraid to message me in discord or message the #Help channel.

But here is the idea behind those functions:
`Charge(Vector2 direction)`
This function gets called a single time, when the button is pressed down, this will cause time to slow allowing the player to ain their weapon, what you need to do in this function is do setup such as instantiating the bullet. The bullet should be at a radius around the player given the direction. Or `position = PlatformerController.instance.transform.position + direction * radius`

`OnAimChange(Vector2 direction)`
This function updates every time the input is changed, and will allow you to change the position of the bullet to match where the player is aiming. It's basically the same code as before with `position = PlatformerController.instance.transform.position + direction * radius`

`Fire(Vector2 direction)`
This function will run a single time when the player releases the button, this is when you actually send the the bullet off, by setting it's velocity or however you want to deal with it.

Isaelus committed:
Merge branch 'RicochetProjectile' of https://github.com/AztecGameLab/LongTermGame5 into RicochetProjectile

Isaelus committed:
Merge branch 'main' into RicochetProjectile

Isaelus committed:
Merge branch 'main' into RicochetProjectile

Isaelus committed:
Merge branch 'main' into RicochetProjectile

Isaelus committed:
Finished Ricochet Projectile

Bullet bounces up to 5 times and disappears when hits an entity.

Commit:
Undoing the GameInputs Change

Kainkun committed:
richohete adjustment

Kainkun reviewed (approved):
great work!
